src/shared/tpm2-util.h

```diff
@@ -7,7 +7,8 @@
 #include "macro.h"
 
 typedef enum TPM2Flags {
-        TPM2_FLAGS_USE_PIN = 1 << 0,
+        TPM2_FLAGS_USE_PIN  = 1 << 0,
+        TPM2_FLAGS_USE_SALT = 1 << 1,
```

Comment:
hmm, the flag is redundant, no? it would suffice to just have the salt+size and pass that around, and if it's NULL we have no salt...

Reply:
It is redundant. I guess it would be a just, check salt in JSON, if not, no salt.

Reply:
yes, i think that would be nicer, let's avoid redundant fields